src-os/ptime_clock_stubs.c

```diff
@@ -181,9 +226,34 @@ CAMLprim value ocaml_ptime_clock_current_tz_offset_s (value unit)
        (dd == -1 || dd >  1 /* year wrap */) ? dm - (24 * 60) :
        dm /* same day */;
 
-  value some = caml_alloc (1, 0);
+  some = caml_alloc (1, 0);
   Store_field (some, 0, Val_int (dm * 60));
-  return some;
+  CAMLreturn(some);
+}
+
+#elif defined(OCAML_PTIME_WIN)
+
+CAMLprim value ocaml_ptime_clock_current_tz_offset_s (value unit)
+{
+  CAMLparam1(unit);
+  CAMLlocal1(some);
+  TIME_ZONE_INFORMATION tz;
+  int bias;
+
+  DWORD ret = GetTimeZoneInformation(&tz);
+  if (ret == TIME_ZONE_ID_UNKNOWN)
+    bias = tz.Bias;
+  else if (ret == TIME_ZONE_ID_STANDARD)
+    bias = tz.Bias + tz.StandardBias;
+  else if (ret == TIME_ZONE_ID_DAYLIGHT)
+    bias = tz.Bias + tz.DaylightBias;
+  else {
+    OCAML_PTIME_RAISE_SYS_ERROR("GetTimeZoneInformation failed");
+  }
+
+  some = caml_alloc (1, 0);
+  Store_field (some, 0, Val_int (-bias * 60));
```

Comment:
It's probably worth a comment noting (from MSDN) that bias is defined using `UTC = localtime + bias` (i.e. the other way around from the Unix implementation above)

Reply:
@IndiscriminateCoding could you please add something along these lines.